frontend/src/components/custom-tools/prompt-card/PromptCardItems.jsx

```diff
@@ -123,6 +109,32 @@ function PromptCardItems({
     return result;
   };
 
+  const getUpdatedCoverage = (promptId, singlePass, promptOutputs) => {
+    let updatedCoverage = null;
+    Object.keys(promptOutputs).forEach((key) => {
+      const [keyPromptId, , , keyIsSinglePass] = key.split("__"); // Destructure the key parts
+
+      // Check if the key matches the criteria
+      if (keyPromptId === promptId && keyIsSinglePass === String(singlePass)) {
+        const currentCoverage = promptOutputs[key]?.coverage || [];
+
+        // Update the highestCoverage if the current one is longer
+        if (
+          !updatedCoverage ||
+          currentCoverage.length > updatedCoverage.length
+        ) {
+          updatedCoverage = currentCoverage;
+        }
+      }
+    });
+
+    return updatedCoverage;
+  };
+
+  const promptCoverage =
+    getUpdatedCoverage(promptId, singlePassExtractMode, promptOutputs) ||
+    coverageCountData;
```

Comment:
@jagadeeswaran-zipstack I think it would be better if we maintain a state to store this value. If not, this logic will be executed everytime the component re-renders.

Reply:
Yes, you're right. we'll improve this in next iteration.